docs/features/kubernetes/authentication.md

```diff
@@ -0,0 +1,26 @@
+---
+id: kubernetes-auth
+title: Kubernetes Authentication
+description: Authentication in kubernetes plugin
+---
+
+The authentication process in kubernetes is basically separate from backstage auth, the
+providers are configured so your kubernetes plugin can locate and access the clusters you
+have access to, the providers currently available are categorized in server side auth and
+client side auth, here's the list:
+
+## Server Side Providers
+
+These providers authenticate your application with the clusters, meaning anyone that is
+logged in into your backstage app will be able to access the same clusters, the providers
+available as server side are `serviceAccount`, `googleServiceAccount`, `aws`, `azure`
+and `localKubectlProxy`.
```

Comment:
```suggestion
available as server side are:
* `aws`
* `azure`
* `googleServiceAccount`
* `localKubectlProxy`
* `serviceAccount`
```
I think an alphabetized bullet list makes it easier to see where to contribute new auth providers when they get added in the future. Also, I didn't figure out how to do it, but maybe each of these could be a link to appropriate anchors in the configuration doc? It would be awesome if I could click `localKubectlProxy` and be taken to a description of what it means.

Reply:
The issue here is that these providers don't map 1:1 to the auth providers of the backstage app, case in point, `azure` would be a client side auth strategy in the app, but a server side auth with the azure CLI in the Kubernetes plugin, so not sure if it's ok to link to the same doc section...

Reply:
I was actually imagining it would somehow be possible to link to the individual table rows under https://backstage.io/docs/features/kubernetes/configuration#clustersauthprovider, but I was mistaken